Adds key resources to `Teleport resources` section (#12418)
gravitational/teleport: docs/pages/access-controls/reference.mdx

you don't want regular users editing SSO (`auth_connector`) or creating and editing new roles
(`role`).

Below is an example `allow` section that illustrates commonly used `rules`. 
Each rule includes a list of Teleport resources and the CRUD 

> operations that a user is allowed to execute on them.:

Reviewer: ```suggestion
operations that a user is allowed to execute on them:
```

```yaml
allow:
  rules:

>     # Role: CRUD options for managing Teleport Server Access Nodes

Reviewer: ```suggestion
    # CRUD options for managing Teleport Server Access Nodes
```

    - resources:
        - node
      verbs: [list, create, read, update, delete]
    - resources:
        - app_server
      verbs: [list, create, read, update, delete]
    - resources:
        - kube_service
      verbs: [list, create, read, update, delete]
    - resources:
        - db_server
      verbs: [list, create, read, update, delete]
    - resources:
        - windows_desktop
      verbs: [list, create, read, update, delete]
    - resources:
        - role
      verbs: [list, create, read, update, delete]

>     # Auth Connectors: aka SSO connectors.

Reviewer: ```suggestion
    # Auth connectors are also known as SSO connectors
```

    - resources:
        - auth_connector
      verbs: [list, create, read, update, delete]
    # Session: Provides access to Session Recordings.
    # e.g If session read is false, users can't play the recordings
    # It is possible to restrict "list" but to allow "read" (in this case a user will
    # be able to replay a session using `tsh play` if they know the session ID).
    - resources:
        - session
      verbs: [list, read]
    - resources:
        - trusted_cluster
      verbs: [list, create, read, update, delete]

>     # Events: Determines whether a user view the audit log and session recordings.

Reviewer: ```suggestion
    # Events: Determines whether a user can view the audit log
    # and session recordings.
```

    - resources:
        - event
      verbs: [list, read]
    - resources:
        - user
      verbs: [list, create, read, update, delete]

>     # Enterprise customers can enable app and node wizard enabling the token resource.

Reviewer: Just double-checking: Does this mean that by enabling these CRUD operations on the token resource, Enterprise customers can enable the app and Node wizard?

    - resources:
        - kube_service
      verbs: [list, create, read, update, delete]
    - resources:
        - db_server
      verbs: [list, create, read, update, delete]
    - resources:
        - windows_desktop
      verbs: [list, create, read, update, delete]
    - resources:
        - role
      verbs: [list, create, read, update, delete]
    # Auth connectors are also known as SSO connectors
    - resources:
        - auth_connector
      verbs: [list, create, read, update, delete]
    # Session: Provides access to Session Recordings.
    # e.g If session read is false, users can't play the recordings
    # It is possible to restrict "list" but to allow "read" (in this case a user will
    # be able to replay a session using `tsh play` if they know the session ID).
    - resources:
        - session
      verbs: [list, read]
    - resources:
        - trusted_cluster
      verbs: [list, create, read, update, delete]
    # Events: Determines whether a user can view the audit log
    # and session recordings.
    - resources:
        - event
      verbs: [list, read]
    - resources:
        - user
      verbs: [list, create, read, update, delete]

>     # Enterprise customers can enable app and node wizard enabling the token resource.

Reviewer: ```suggestion
```
I think deleting the line should also be okay, since it doesn't seem specific to the configuration below it.

you don't want regular users editing SSO (`auth_connector`) or creating and editing new roles
(`role`).

> Below is an example `allow` section of commonly used `rules`:

Reviewer: ```suggestion
Below is an example `allow` section that illustrates commonly used `rules`:
```

```yaml
allow:

>   rules:

Reviewer: Each resource-specific comment includes mostly the same text. Could we make this a bit cleaner by extracting these comments into another paragraph above the sample YAML?

E.g.,

```md
Each rule includes a list of Teleport resources and the CRUD 
operations that a user is allowed to execute on them.
``` 
We could then use comments in the YAML snippet to clarify what each `resources` value means if it's not explicit enough.

Author: Yeah good idea, this is unnecessarily repetitive.
